src/components/app/ServiceWorkerManager.js

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import React, { PureComponent } from 'react';
+import explicitConnect from '../../utils/connect';
+
+import { getDataSource } from '../../reducers/url-state';
+import { getView } from '../../reducers/app';
+
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+import type { DataSource } from '../../types/actions';
+import type { Phase } from '../../types/reducers';
+
+import './ServiceWorkerManager.css';
+
+type StateProps = {|
+  +dataSource: DataSource,
+  +phase: Phase,
+|};
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+type InstallStatus = 'pending' | 'ready' | 'idle';
+type State = {|
+  installStatus: InstallStatus,
+  closeNotice: boolean,
+|};
+
+class ServiceWorkerManager extends PureComponent<Props, State> {
+  state = {
+    installStatus: 'idle',
+    closeNotice: true,
```

Comment:
I was surprised that this state was `true` at first. But after reading the code, it made sense. But it still feels like `displayNotice` would be better for that with the inverted values. What do you think?

Reply:
I don't mind :) Also maybe make it look like more as a boolean: isNoticeDisplayed?

Reply:
Yeah, `isNoticeDisplayed` seems good to me.